File: Project 1 Phase 4/meam620-kmel/meam620-kmel/QC-12.12.12/interfaces/kQuadIface-12.12.12/testSendQuadCmd4.m
%Copyright KMel Robotics 2012. Must read KMEL_LICENSE.pdf for terms and conditions before use.
function testSendQuadCmd4()

%open the interface
KQUAD.dev    = '/dev/ttyUSB0';
KQUAD.driver = @kQuadInterfaceAPI;
KQUAD.baud   = 1000000;
KQUAD.id     = 1;
KQUAD.chan   = 1;
KQUAD.type   = 0; %0 for standard, 1 for nano

KQUAD.driver('connect',KQUAD.dev,KQUAD.baud)

while(1)
  
  thrust = 1; %grams. 1 for idle
  roll   = 0; %radians
  pitch  = 0; %radians
  yaw    = 0; %radians
  
  trpy = [thrust roll pitch yaw];
  
  drpy = [0 0 0];     %angular rates (rad/sec)
  kp   = [0 0 0];     %proportional gains
  kd   = [0 0 0];     %derivative gains
  ms   = [0 0 0];     %moments (x,y,z)

  KQUAD.driver('SendQuadCmd4',KQUAD.id, KQUAD.chan, KQUAD.type, ...
                trpy, drpy, kp, kd, ms);

  pause(0.03);
end

File: Project 1 Phase 4/meam620-kmel/meam620-kmel/QC-12.12.12/interfaces/kQuadIface-12.12.12/testSendQuadCmd4.cc
/* Copyright KMel Robotics 2012. Must read KMEL_LICENSE.pdf for terms and conditions before use. */
#include "kQuadInterface.hh"

int main()
{
  char * dev = (char*)"/dev/ttyUSB0";
  int baud   = 1000000;
  
  kQuadInterface kqi;
  
  if (kqi.Connect(dev,baud))
  {
    printf("could not connect to the device\n");
    return -1;
  }
  
  if (kqi.StartSendThread())
  {
    printf("could not start thread\n");
    return -1;
  }
  
  uint8_t quadId   = 1;
  uint8_t quadType = 0; //standard
  uint8_t channel  = 1;
  
  float thrust = 1; //(grams)1 for idle
  float roll   = 0; //radians
  float pitch  = 0; //radians
  float yaw    = 0; //radians
  
  float droll  = 0;
  float dpitch = 0;
  float dyaw   = 0;
  
  float kp_roll   = 550;
  float kp_pitch  = 550;
  float kp_yaw    = 800;
  
  float kd_roll   = 42;
  float kd_pitch  = 42;
  float kd_yaw    = 50;
  
  float mx        = 0;  //extra moments (in grams)
  float my        = 0;
  float mz        = 0;
  
  while(1)
  {
    kqi.SendQuadCmd4(quadId, quadType, channel, thrust, roll, pitch, yaw,
                     droll,dpitch,dyaw, kp_roll,kp_pitch,kp_yaw, kd_roll,kd_pitch,kd_yaw,mx,my,mz);
    
    usleep(50000);
    
    printf("."); fflush(stdout);
  }
  
 
  return 0;
}